sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/PipelinedDocumentQueryExecutionContext.java

```diff
@@ -53,12 +55,17 @@ private static BiFunction<String, PipelinedDocumentQueryParams<Document>, Flux<I
             createBaseComponentFunction = (continuationToken, documentQueryParams) -> {
                 CosmosQueryRequestOptions orderByCosmosQueryRequestOptions =
                     qryOptAccessor.clone(requestOptions);
-                ModelBridgeInternal.setQueryRequestOptionsContinuationToken(orderByCosmosQueryRequestOptions, continuationToken);
-                qryOptAccessor.getImpl(orderByCosmosQueryRequestOptions).setCustomItemSerializer(null);
-
-                documentQueryParams.setCosmosQueryRequestOptions(orderByCosmosQueryRequestOptions);
-
-                return OrderByDocumentQueryExecutionContext.createAsync(diagnosticsClientContext, client, documentQueryParams, collection);
+                if (queryInfo.hasNonStreamingOrderBy()) {
+                    checkNotNull(continuationToken, "Can not use a continuation token for a vector search query ");
```

Comment:
So `continuationToken` is not necessary for non-streaming order by query, yet we do `null` check here - `checkNotNull` will throw if `continuationToken` is `null` - am I misunderstanding something?

Reply:
Fixed the code, we need to check if somehow the customer passed in continuation token with the vector search query.